packages/forms/frontend/sirius-components-widget-reference/src/components/ModelBrowserTreeView.tsx

```diff
@@ -43,33 +44,47 @@ export const ModelBrowserTreeView = ({
 }: ModelBrowserTreeViewProps) => {
   const { classes } = useTreeStyle();
 
-  const [state, setState] = useState<ModelBrowserTreeViewState>({ filterBarText: '' });
+  const [state, setState] = useState<ModelBrowserTreeViewState>({
+    filterBarText: '',
+    expanded: [],
+    maxDepth: 1,
+  });
+
+  const treeId: string = `modelBrowser://${leafType}?ownerKind=${encodeURIComponent(
+    ownerKind
+  )}&targetType=${encodeURIComponent(widget.reference.referenceKind)}&ownerId=${
+    widget.ownerId
+  }&descriptionId=${encodeURIComponent(widget.descriptionId)}&isContainment=${widget.reference.containment}`;
+  const { tree } = useModelBrowserSubscription(editingContextId, treeId, state.expanded, state.maxDepth);
+
+  const onExpandedElementChange = (expanded: string[], maxDepth: number) => {
+    setState((prevState) => ({ ...prevState, expanded, maxDepth }));
+  };
 
   return (
     <>
       <ModelBrowserFilterBar
-        onTextChange={(event) => setState({ filterBarText: event.target.value })}
-        onTextClear={() => setState({ filterBarText: '' })}
+        onTextChange={(event) => setState((prevState) => ({ ...prevState, filterBarText: event.target.value }))}
+        onTextClear={() => setState((prevState) => ({ ...prevState, filterBarText: '' }))}
         text={state.filterBarText}
       />
       <span className={classes.title}>{title}</span>
       <div className={classes.borderStyle}>
-        <TreeView
-          editingContextId={editingContextId}
-          readOnly={true}
-          treeId={`modelBrowser://${leafType}?ownerKind=${encodeURIComponent(
-            ownerKind
-          )}&targetType=${encodeURIComponent(widget.reference.referenceKind)}&ownerId=${
-            widget.ownerId
-          }&descriptionId=${encodeURIComponent(widget.descriptionId)}&isContainment=${widget.reference.containment}`}
-          enableMultiSelection={enableMultiSelection}
-          synchronizedWithSelection={true}
-          activeFilterIds={[]}
-          textToFilter={state.filterBarText}
-          textToHighlight={state.filterBarText}
-          markedItemIds={markedItemIds}
-          treeItemActionRender={(props) => <WidgetReferenceTreeItemAction {...props} />}
-        />
+        {tree && (
```

Comment:
Don't use this way for conditional rendering, be explicit because the implicit casting in JavaScript for boolean is funky